### PR TITLE
New contributions guidelines

### LINKED ISSUE #2194: Guidelines for new contributions
New contributions are always welcome. However, ML is a rapidly changing field. There are hundreds of papers being published on `arXiv` everyday. With so many papers coming out, we expect new things that aren't readily available in the `core` and people would be wiling to contribute to `addons`. 

From the perspective of an end user, it makes sense that `addons` incorporate new contributions as much as possible. However from the perspective of a maintainer, things can become overwhelming at some point in time because all contributors won't be active all the time which can lead to stalled issues and problems with new updates/releases (if any).

To this end, we need to decide how many `citations` should be there for a new functionality to be added in the ecosystem. For the `core`, it's around 100 citations IIRC. For addons, this bar would be much lower but we still need to decide a number. We should update this in the `README` as well. 

There are certain ways to track the number of citations but one of the coolest tools is this one: https://github.com/dennybritz/papergraph

### PR DESCRIPTION
**Pull request by Harsh188**
# Description

Adding two new guidelines to CONTRIBUTING.md
Fixes #2194

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by bhack** [comment]
We need to finalize the discussion at https://github.com/tensorflow/addons/issues/2194. I suggest to put this in the next meeting agenda /cc @seanpmorgan

**Comment by Harsh188** [comment]
@seanpmorgan @bhack Could you guys kindly take another look at this PR? Thanks.

**Comment by Harsh188** [comment]
I think I should mention that compositional ops are preferred over custom ops.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,6 +43,13 @@ fitting the following criteria:
  * Lastly, the functionality conforms to the contribution guidelines of
  its subpackage.
 
+Suggested guidelines for new feature requests:
+
+* The feature contains an official refrence implementation.
```
Comment (AakashKumarNain) [inline CONTRIBUTING.md:48]: Reference*

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,6 +43,13 @@ fitting the following criteria:
...
+* Should be able to reproduce the same results in a published paper.
+* Should provide out-of-box performance.
```
Comment (AakashKumarNain) [inline CONTRIBUTING.md:50]: Not very clear to the end user. What does it mean?

Comment (Harsh188) [reply CONTRIBUTING.md:50]: I guess what I'm trying to articulate here is that the feature is recommended to contain a relevant DL academic paper. In addition, the feature being implemented should be able to perform with similar results from that of the relevant DL academic paper.

Comment (AakashKumarNain) [reply CONTRIBUTING.md:50]: I was pointing out this point specifically `Should provide out-of-box performance.`

Comment (Harsh188) [reply CONTRIBUTING.md:50]: Great! Thanks for the suggestions. I apologize for my untimely responses since I've been busy studying for my midterms 😢.

Comment (AakashKumarNain) [reply CONTRIBUTING.md]: No worries. Thanks for the contribution

**Comment by AakashKumarNain** [comment]
@bhack @seanpmorgan I think this is good to go. We can later write a bot to automate this process using `scholarly`

**Review by AakashKumarNain — DISMISSED** [review]

**Comment by bhack** [comment]
Can we add a similar notice in our fature request Issue Template?

**Comment by Harsh188** [comment]
> @bhack @seanpmorgan I think this is good to go. We can later write a bot to automate this process using `scholarly`
@AakashKumarNain 
I would love to work on this. I'm looking for project ideas for GSoC 2021. Do you think this would be an appropriate project proposal? Please feel free to give me advise/suggestions. I've posted a message on Gitter and I'm awaiting responses.

> Can we add a similar notice in our fature request Issue Template?
I've added two more checks:
1. To check if the user has read `CONTRIBUTING.md`.
2. To check if the academic paper exceeds 50 citations.

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 3]
```diff
@@ -7,9 +7,11 @@ about: Use this template for raising a feature request
 **Describe the feature and the current behavior/state.**
 
 **Relevant information**
+- Have you read the requirements for new contributions in [`CONTRIBUTING.md`](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#requirements-for-new-contributions-to-the-repository)? (yes/no):
```
Comment (bhack) [inline .github/ISSUE_TEMPLATE/feature-request.md:10]: I suppose something more like:
If you want to contribute read the requirements...

Comment (Harsh188) [reply .github/ISSUE_TEMPLATE/feature-request.md:10]: I'll include it as a note after the "Are you willing to contribute" bullet point.

**Review by bhack — APPROVED** [review]

**Pull request merged**